server/task_executor_data_update.go

```diff
@@ -4,18 +4,20 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"sync/atomic"
 
 	"github.com/bytebase/bytebase/api"
 	"github.com/bytebase/bytebase/plugin/db"
 )
 
 // NewDataUpdateTaskExecutor creates a data update (DML) task executor.
 func NewDataUpdateTaskExecutor() TaskExecutor {
-	return &SchemaUpdateTaskExecutor{}
+	return &DataUpdateTaskExecutor{}
 }
 
 // DataUpdateTaskExecutor is the data update (DML) task executor.
 type DataUpdateTaskExecutor struct {
+	completed int32
 }
 
 // RunOnce will run the data update (DML) task executor once.
```

Comment:
`defer` atomically set `completed` flag at the start of `RunOnce`

Reply:
Done.